src/Context/GlobalContext/StateContext.jsx

```diff
@@ -0,0 +1,72 @@
+import { createContext, useContext, useReducer, useEffect, useState } from "react"
+import axios from "axios"
+import { uniqueCategory } from "../../utils/uniqueCategory"
+import { sortByPrice } from "../../utils/sortByPrice"
+import { stateReducerFun } from "../GlobalContext/StateReducer"
+
+const StateContext = createContext()
+
+
+const initialState = {
+    products: [],
+    filters: {
+        sortBy: '',
+        priceRange: 20,
+        categoryName: [],
+        rating: ''
+    },
+    wishlist: [],
+}
```

Comment:
Just a suggestion but you can have more than one logically separated context as well. 
For example, You can create a different context for filters and wishlist.